### PR TITLE
update jasmine testing

### PR DESCRIPTION
**Pull request by Trewaters**
started but didn't finish creating a testing topic. I could finish it over the next few days.

Please look at it too because I wasn't sure if it was okay to combine Jasmine/Mocha/Chai into the same lesson. The title of the issue says Jasmine but the bullet points are about Mocha/Chai.

Your comments are welcome!

### CODE REVIEW

File: testing-and-tdd/jasmine-testing.md [hunk 1]
```diff
@@ -12,51 +12,128 @@ Here are links to lessons that should be completed before this lesson:
...
+
+**1st test**
+
+`describe` and `it` functions work similar to jasmine's same named functions.  
+```javascript
+var assert = require('assert');
+
+describe('Mocha String Test', function () {
+ it('should return the exact number of characters in a string', function () {
+        assert.equal("Hello".length, 4);
```
Comment (alodahl) [inline testing-and-tdd/jasmine-testing.md]: "Hello".length == 4 ?

Comment (Trewaters) [reply testing-and-tdd/jasmine-testing.md]: Showing what a fail looks like in Mocha but I think I should also add a comment to that so it is clear that is what's going on.

**Comment by Trewaters** [comment]
Thank you for that comment. Always good to have more than one pair of eyes looking at things.

**Comment by Trewaters** [comment]
Sorry I ran into delays working on this. I am hoping to have it completed by Saturday morning at the latest, hopefully late Friday if things go my way ;)

**Comment by alodahl** [comment]
Thank you Tre!

**Comment by alodahl** [comment]
Tre, what you have so far is perfect. Would you be willing to work on it further in the near future? If so, how much time do you think you’d need?

**Comment by Trewaters** [comment]
Thank you!

I would be willing to work on it more. I could finish up the rest by Wednesday if that works for you.

Tre'

Responding from my cellphone...

________________________________

**Comment by alodahl** [comment]
That would be great!  
Regarding your comments:
- Your tutorials are really good, and often the slideshows are just visuals from the lesson portion of the outline.  Explaining the same things, but with images of your file tree, tests, and command line dialog could be really helpful for visual learners.
- We really do need Independent Practice and a Check for Understanding.  You have some good links listed here - you could pick one or a few of them to assign to the apprentices for Independent Practice, then list the rest as "supplemental resources".
- The check for understanding is just a way for apprentices to self-assess their knowledge so they know what to study for an actual assessment.  They've learned basic js, jquery, and object oriented programming just a few days before, so maybe you could challenge them to meet the 'objectives' points for each type of test.

**Comment by alodahl** [comment]
That web dev journey video cracked me up, btw.

**Comment by Trewaters** [comment]
Here is a link to the slides I made for this unit ( https://docs.google.com/presentation/d/1jMkVbUkZS40z7kAoC1mPIt8RwfF8-GRL8oeee25UzYw/edit?usp=sharing ). 

I also finished the Independent Practice, Challenge, and check for understanding sections.

Please review for clarity.

File: testing-and-tdd/jasmine-testing.md [hunk 2]
```diff
@@ -12,67 +12,176 @@ Here are links to lessons that should be completed before this lesson:
...
 
-Apprentices can try to do this other thing.
+*Instructions:* Make your independent practice test run in the browser. The challenge is the environment must be modified for this test to run in the browser instead of the command line.
 
 
 ### Check for Understanding
 
-Some ideas: have apprentices summarize to each other, make a cheat sheet, take a quiz, do an assignment, or something else that helps assess their understanding.
+*Question:* What is Jasmine?
+*Answer to Q:* Jasmine is a behavior-driven development (BDD) for testing JavaScript code. Jasmine has no external dependencies and does not require a DOM.
```
Comment (alodahl) [inline testing-and-tdd/jasmine-testing.md]: Maybe have this elsewhere in the curriculum, that way they're not looking right at it as they answer?

**Comment by Trewaters** [comment]
This means a lot. Next time I will do better to stay on deadline.

File: testing-and-tdd/jasmine-testing.md (hunk 2 above)
Comment (Trewaters) [reply testing-and-tdd/jasmine-testing.md]: I moved this to the top for when the students are first introduced to Jasmine. Plus I changed the question a little.

*Question:* What is Jasmine and how does it work in your project?

**Comment by Trewaters** [comment]
I changed the slide ownership. You should be able to control it now.

**Pull request merged**